packages/context/src/inject.ts

```diff
@@ -167,13 +187,20 @@ export namespace inject {
   };
 }
 
-function resolveAsGetter(ctx: Context, injection: Injection) {
+function resolveAsGetter(
+  ctx: Context,
+  injection: Injection,
+  session?: ResolutionSession,
+) {
+  // We need to clone the session for the getter as it will be resolved later
+  session = ResolutionSession.folk(session);
```

Comment:
I find the name `folk` confusing, did you perhaps mean `fork`?

I was not able to find any dictionary entry for `folk` as a verb, it's typically used as a noun or adjective, see e.g. https://www.thefreedictionary.com/folk and https://dictionary.cambridge.org/dictionary/english/folk